src/dev/riscv/plic.cc

```diff
@@ -421,6 +429,37 @@ Plic::propagateOutput()
     }
 }
 
+void
+Plic::initContextFromNContexts(int n_contexts)
+{
+    for (uint32_t i = 0; i < (uint32_t)n_contexts; i += 2) {
+      contextConfigs.emplace_back((i >> 1), ExceptionCode::INT_EXT_MACHINE);
+      contextConfigs.emplace_back((i >> 1), ExceptionCode::INT_EXT_SUPER);
+    }
+}
+
+void
+Plic::initContextFromHartConfig(const std::string& hart_config)
+{
+    uint32_t hart_id = 0;
+    for (char c: hart_config) {
+      switch (c) {
+        case ',':
+          hart_id++;
+          break;
+        case 'M':
+          contextConfigs.emplace_back(hart_id, ExceptionCode::INT_EXT_MACHINE);
+          break;
+        case 'S':
+          contextConfigs.emplace_back(hart_id, ExceptionCode::INT_EXT_SUPER);
+          break;
+        default:
+          panic("hart_config should not contains the value: %c", c);
```

Comment:
```suggestion
          fatal("hart_config should not contains the value: %c", c);
```

This is a config issue, not a bug in gem5.